doc/contributing.rst

```diff
@@ -0,0 +1,809 @@
+
+   ((xarray) $ pytest test_cool_feature.py -v
+    =============================== test session starts ================================
+    platform darwin -- Python 3.6.4, pytest-3.2.1, py-1.4.34, pluggy-0.4.0 --
+    cachedir: ../../.cache
+    plugins: cov-2.5.1, hypothesis-3.23.0
+    collected 11 items
+
+    test_cool_feature.py::test_dtypes[int8] PASSED
+    test_cool_feature.py::test_dtypes[int16] PASSED
+    test_cool_feature.py::test_dtypes[int32] PASSED
+    test_cool_feature.py::test_dtypes[int64] PASSED
+    test_cool_feature.py::test_mark[float32] PASSED
+    test_cool_feature.py::test_mark[int16] SKIPPED
+    test_cool_feature.py::test_mark[int32] xfail
+    test_cool_feature.py::test_series[int8] PASSED
+    test_cool_feature.py::test_series[int16] PASSED
+    test_cool_feature.py::test_series[int32] PASSED
+    test_cool_feature.py::test_series[int64] PASSED
+
+    ================== 9 passed, 1 skipped, 1 xfailed in 1.83 seconds ==================
+
+Tests that we have ``parametrized`` are now accessible via the test name, for
+example we could run these with ``-k int8`` to sub-select *only* those tests
+which match ``int8``.
+
+
+.. code-block:: shell
+
+   ((xarray) bash-3.2$ pytest  test_cool_feature.py  -v -k int8
+   =========================== test session starts ===========================
+   platform darwin -- Python 3.6.2, pytest-3.2.1, py-1.4.31, pluggy-0.4.0
+   collected 11 items
+
+   test_cool_feature.py::test_dtypes[int8] PASSED
+   test_cool_feature.py::test_series[int8] PASSED
+
+
+Running the test suite
+----------------------
+
+The tests can then be run directly inside your Git clone (without having to
+install *xarray*) by typing::
+
+    pytest xarray
+
+The tests suite is exhaustive and takes a few minutes.  Often it is
+worth running only a subset of tests first around your changes before running the
+entire suite.
+
+The easiest way to do this is with::
+
+    pytest xarray/path/to/test.py -k regex_matching_test_name
+
+Or with one of the following constructs::
+
+    pytest xarray/tests/[test-module].py
+    pytest xarray/tests/[test-module].py::[TestClass]
+    pytest xarray/tests/[test-module].py::[TestClass]::[test_method]
+
+Using `pytest-xdist <https://pypi.python.org/pypi/pytest-xdist>`_, one can
+speed up local testing on multicore machines. To use this feature, you will
+need to install `pytest-xdist` via::
+
+    pip install pytest-xdist
+
+
+Then, run pytest with the optional -n argument:
+
+    pytest xarray -n 4
+
+This can significantly reduce the time it takes to locally run tests before
+submitting a pull request.
+
+For more, see the `pytest <http://doc.pytest.org/en/latest/>`_ documentation.
+
+Running the performance test suite
+----------------------------------
+
+Performance matters and it is worth considering whether your code has introduced
+performance regressions.  *xarray* is in the process of migrating to
+`asv benchmarks <https://github.com/spacetelescope/asv>`__
```

Comment:
Instead of saying that we're in the process of migrating to asv, say that we're starting to write asv tests. I don't think we're migrating from anything else :)